pkg/network/protocols/http/shared_libraries.go

```diff
@@ -178,6 +198,44 @@ func (w *soWatcher) sync() {
 	}
 }
 
+func (w *soWatcher) checkProcessDone() (updated bool) {
```

Comment:
I don't understand the use of `pathsToBeRemoved`
you are just filling it without ever using it
furthermroe, the entire first loop is redundant, as you are always unregistering all hooks on line 232

Reply:
Thanks, it has been fixed here https://github.com/DataDog/datadog-agent/pull/13803/commits/6058ef9f48a9835e2783012fab24811a4e2c061c